assets/js/components/modules-list.js

```diff
@@ -78,10 +78,15 @@ class ModulesList extends Component {
 	}
 
 	render() {
+		const { moduleSlugs } = this.props;
 		const modulesData = getModulesData();
 
-		// Filter out internal modules.
-		const modules = Object.values( modulesData ).filter( ( module ) => ! module.internal );
+		// Filter out internal modules and specific ones via the moduleSlugs prop.
+		const modules = Object.values( modulesData ).filter(
+			( module ) =>
+				! module.internal &&
+				( Array.isArray( moduleSlugs ) && moduleSlugs.length > 0 ? moduleSlugs.includes( module.slug ) : true )
```

Comment:
This works, but is a bit much to have all on one line. It also checks the type an length of `moduleSlugs` on every iteration which is unnecessary. The `true` at the end of the ternary is also a bit confusing at first glance. Of course it doesn't make a huge difference with such a small list, but we can make it a bit easier to read by separating these things.

Rather than doing everything in the filter for every module, we can change how module objects are selected before filtering based on `moduleSlugs`:

```js
const moduleObjects = Array.isArray( moduleSlugs ) && moduleSlugs.length
    ? moduleSlugs
        .filter( ( slug ) => modulesData.hasOwnProperty( slug ) )
        .reduce( ( acc, slug ) => ( { ...acc, [ slug ]: modulesData[ slug ] } ), {} )
    : modulesData
```

This way `moduleObjects` come from `moduleSlugs` if it's passed as an array, otherwise it uses them all like before.

Then we only need to use `Object.values( moduleObjects )` instead in the existing line which would otherwise remain unchanged to filter out internal modules in all cases.

What do you think?

Reply:
Makes sense @aaemnnosttv . I've updated the PR, can you take a look again?
Also I was thinking maybe we shouldn't check if the module is an internal one if moduleSlugs is not empty. The reason being if we have something in moduleSlugs, we intentionally want to display that module. So I was thinking whether we can do something like this:
`
if( Array.isArray( moduleSlugs ) && moduleSlugs.length ) {
    // Filter per slugs.
}
else {
   // Filter internal modules.
}
`

What do you think?

Reply:
> Also I was thinking maybe we shouldn't check if the module is an internal one if moduleSlugs is not empty

Good question – I had the same thought 😄  `internal` modules are those without any interface/settings (the only one right now is `site-verification` I think). So it wouldn't make sense to display an `internal` module in this context, but otherwise it's a very reasonable suggestion.